server/src/server.ts

```diff
@@ -54,14 +54,16 @@ export default class GameServer {
     
         if(roomToJoin.gameStarted) { 
             socket.emit('joinRoom', false);
-        } else {
+        } else if (Object.keys(roomToJoin.getPlayers()).length < roomToJoin.maxCapacity) {
```

Comment:
we should move this into a small function in room (similar to isEmpty). then you can just call room.isFull()  in this check